### PR TITLE
Improved The Performance Of Logger

### PR DESCRIPTION
**Pull request by noumanurrahman**
Updated `LoggerExtensions.cs` To Only Even Try To Log If The Log Level Is Enable To Improve Performance Of Logger By Saving Memory Allocation

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=66895) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=66895) before we can accept your contribution.<br/><br/>:x: ProgrammingFire [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=66895)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=66895) it.</sub>

File: src/libraries/Microsoft.Extensions.Logging.Abstractions/src/LoggerExtensions.cs [hunk 1]
```diff
@@ -25,7 +25,10 @@ public static class LoggerExtensions
         /// <example>logger.LogDebug(0, exception, "Error while processing request from {Address}", address)</example>
         public static void LogDebug(this ILogger logger, EventId eventId, Exception? exception, string? message, params object?[] args)
         {
-            logger.Log(LogLevel.Debug, eventId, exception, message, args);
+            if (logger.IsEnabled(LogLevel.Debug))
```
Comment (gfoidl) [inline src/libraries/Microsoft.Extensions.Logging.Abstractions/src/LoggerExtensions.cs:28]: Which allocations should be saved by this pattern?

The `logger.IsEnabled`-check should be done before the call to `LogDebug` in user code.
Thus the allocation of `args` (the array) and potentially the creation of `message` can be saved. These objects are already existent when reaching the check you introduced here.

Or even better use [LoggerMessage.Define](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loggermessage.define?view=dotnet-plat-ext-6.0) where that behavior can be set / use the (new) logging source generator.

**Comment by tarekgh** [comment]
@ProgrammingFire please provide the benchmark numbers for the allocation and the code you are using to measure it.

**Comment by ProgrammingFire** [comment]
Check this new commit which actually improves the memory allocation by not storing the first three arguments into objects so that they are not boxed and unboxed that saves memory allocation

**Comment by ProgrammingFire** [comment]
I think the improvements in the logger depends on the user that's why I am closing this pull request.

**Pull request closed without merging**